src/main.rs

```diff
@@ -162,23 +162,18 @@ enum Command {
         #[structopt(short, long, parse(from_os_str))]
         target_dir: Option<PathBuf>,
     },
-    /// Compiles the smart contract
+    /// Compiles the contract, generates metadata, packs both together in one <package-name>.contract file
     #[structopt(name = "build")]
     Build {
         /// Path to the Cargo.toml of the contract to build
         #[structopt(long, parse(from_os_str))]
         manifest_path: Option<PathBuf>,
-        #[structopt(flatten)]
-        verbosity: VerbosityFlags,
-        #[structopt(flatten)]
-        unstable_options: UnstableOptions,
-    },
-    /// Generate contract metadata artifacts
-    #[structopt(name = "generate-metadata")]
-    GenerateMetadata {
-        /// Path to the Cargo.toml of the contract for which to generate metadata
-        #[structopt(long, parse(from_os_str))]
-        manifest_path: Option<PathBuf>,
+        /// Only the Wasm and the metadata are generated, no packed .contract file is created
+        #[structopt(long = "skip-packing", conflicts_with = "skip-metadata")]
+        skip_packing: bool,
+        /// Only the Wasm is created, generation of metadata and a packed .contract file is skipped
+        #[structopt(long = "skip-metadata", conflicts_with = "skip-packing")]
+        skip_metadata: bool,
```

Comment:
Can we please name things positively instead of negatively? I.e. `bundle` or `bundle_contract` or `build_bundle` instead of `skip_bundle`? Same for `skip_metadata`.

Reply:
While I agree with the general sentiment of prioritizing positive naming, I'm not sure how we can do that here, since default behavior is wasm+metadata+bundle and we want to provide a possibility to leave certain things out.

So we need better flag names for 
```
--skip-metadata    Only the Wasm is created, generation of metadata and a packed .contract file is skipped
--skip-packing     Only the Wasm and the metadata are generated, no packed .contract file is created
```

Or do you suggest changing default behavior?

Reply:
Why do we need these flags in the first place? I think this should be automated just like you don't need to specify to `cargo` which rust files to build.

Reply:
If your contract implementation has changed you always want to build the Wasm AND if your interfaces have changed you always want to build metadata. This already eliminates the `wasm` flag because bundling pretty much always involved building the `wasm`. So the only question is: Do we want to always build `metadata`? Do we want to infer it? And if so, how? Or do we want to make it skip-opt-in as proposed so far?

Reply:
I don't follow, can you elaborate?

We changed default behavior of `cargo contract build` to generate wasm+metadata+pack. And now we want to provide flags to skip certain steps. Which part of that can be automated?

Reply:
Btw.: Changing of interfaces is done whenever
- message or constructor signature changes
- docs changing (!)
- types changing, even if it is only a type alias renaming to the same type (!)
- pre- or post conditions are changing
- storage is changing

So in pretty much the same cases as "contract is changing" unless the change is clearly not visible to the outside - e.g. kept in the implementation.

Reply:
> I don't follow, can you elaborate?
> 
> We changed default behavior of `cargo contract build` to generate wasm+metadata+pack. And now we want to provide flags to skip certain steps. Which part of that can be automated?

Ah, I was not aware of this. This obviously changes perception of this change to me. Hmm...

Reply:
What is the motivation to change behaviour of `cargo contract build` instead of adding yet another command `cargo contract bundle`?

Reply:
I estimate that those commands will be used most frequently:

- `cargo contract build` (very often)
- `cargo contract bundle` (rarely)
- `cargo contract metadata` (nearly never)

So we could also remove `cargo contract metadata` and add a `--skip-metadata` flag to `cargo contract bundle` and keep `cargo contract build`.

Reply:
>What is the motivation to change behaviour of cargo contract build instead of adding yet another command cargo contract bundle?

See earlier discussion in this PR: https://github.com/paritytech/cargo-contract/pull/97#pullrequestreview-523161594

Reply:
>So we could also remove cargo contract metadata and add a --skip-metadata flag to cargo contract bundle and keep cargo contract build. Building only metadata seldomly makes real sense.

So what this PR already does is that it removes the `cargo contract generate-metadata` cmd, makes `cargo contract build` build everything by default, and adds skip flags for omitting metadata/bundling.

Reply:
Then using the most frequently used command forces users to type in `cargo contract build --skip-metadata` which is messy imo. Generating metadat by default is bad here because it adds tons of overhead and users will question ink! with taking 2 minutes to do a fresh build for a hello world contract. We should focus to make the most frequent command the simplest to use:

- Most frequent: `cargo contract build`
- Less frequent: `cargo contract bundle` or `cargo contract bundle --skip-metadata`
- Least frequent: `cargo contract generate-metadata`

Reply:
Building a bundle without updating the `.wasm` is pretty much always not what you want.